optaplanner-examples/src/main/java/org/optaplanner/examples/nurserostering/optional/score/NurseRosteringConstraintProvider.java

```diff
@@ -228,36 +243,39 @@ Constraint consecutiveWorkingWeekends(ConstraintFactory constraintFactory) {
     Constraint startOnNotFirstDayOfWeekend(ConstraintFactory constraintFactory) {
         return constraintFactory.from(BooleanContractLine.class)
                 .filter(booleanContractLine -> booleanContractLine.getContractLineType() == ContractLineType.COMPLETE_WEEKENDS
-                        &&
-                        booleanContractLine.isEnabled())
-                .join(ShiftAssignment.class, Joiners.equal(ContractLine::getContract, ShiftAssignment::getContract))
-                .groupBy((contract, shift) -> shift.getEmployee(), (contract, shift) -> contract,
+                        && booleanContractLine.isEnabled())
+                .join(ShiftAssignment.class,
+                        Joiners.equal(ContractLine::getContract, ShiftAssignment::getContract))
+                .groupBy((contract, shift) -> shift.getEmployee(),
+                        (contract, shift) -> contract,
                         ExperimentalConstraintCollectors.consecutive((contract, shift) -> shift.getShiftDate(),
                                 ShiftDate::getDayIndex))
                 .flattenLast(ConsecutiveData::getConsecutiveSequences)
-                .filter((employee, contract, shiftList) -> new EmployeeConsecutiveAssignmentStart(
-                        employee, shiftList.getItems().first()).isWeekendAndNotFirstDayOfWeekend())
+                .filter((employee, contract, shiftList) -> isWeekendAndNotFirstDayOfWeekend(employee,
+                        shiftList.getFirstItem()))
                 .penalize("startOnNotFirstDayOfWeekend", HardSoftScore.ONE_SOFT,
                         (employee, contract,
-                                shiftList) -> new EmployeeConsecutiveAssignmentStart(employee, shiftList.getItems().first())
-                                        .getDistanceToFirstDayOfWeekend() * contract.getWeight());
+                                shiftList) -> getDistanceToFirstDayOfWeekend(employee, shiftList.getFirstItem())
+                                        * contract.getWeight());
```

Comment:
No need to create instances over and over again. Have some respect for people's memory use. ;-)

Reply:
Also, this is where the first/last item cache will be marginally useful.